content/automate/ManageIQ/System/Event/StateMachines/Refresh.class/__methods__/check_refreshed.rb

```diff
@@ -0,0 +1,61 @@
+#
+# Description: This method checks the refresh completion.
+#
+
+module ManageIQ
+  module Automate
+    module System
+      module Event
+        module StateMachines
+          module Refresh
+            class CheckRefreshed
+              STATE_FINISHED = 'Finished'.freeze
+              STATUS_OK = 'Ok'.freeze
+
+              def initialize(handle = $evm)
+                @handle = handle
+              end
+
+              def main
+                check_status(refresh_task)
+              end
+
+              private
+
+              def check_status(task)
+                case task.state
+                when STATE_FINISHED
+                  @handle.root['ae_result'] = task.status == STATUS_OK ? 'ok' : 'error'
+                else
+                  @handle.root['ae_result'] = 'retry'
+                  @handle.root['ae_retry_interval'] = '1.minute'
+                end
+              end
+
+              def refresh_task
+                task_id = @handle.get_state_var(:refresh_task_id).try(:first)
+                @handle.log(:info, "Stored refresh task ID: [#{task_id}]")
+                if task_id.nil?
+                  @handle.log(:error, 'Refresh task id is nil')
+                  raise 'Refresh task id is nil'
+                end
+                fetch_task(task_id)
+              end
+
+              def fetch_task(task_id)
+                @handle.vmdb(:miq_task).find(task_id)
+              rescue MiqAeException::ServiceNotFound
```

Comment:
@mkanoor @tinaafitz @gmcculloug Please review this change.
``` @handle.vmdb(:miq_task).find(task_id)``` would throw an exception instead of returning nil when task is not found.

Reply:
Use `@handle.vmdb(:miq_task).find_by(:id => task_id)`